src/main/java/com/google/flourbot/api/Bot.java

```diff
@@ -79,9 +80,17 @@ public void onEvent(@RequestBody JsonNode event) throws IOException, GeneralSecu
         String threadId = event.at("/message/thread/name").asText();
         
         ChatResponse chatResponse = macroExecutionModule.execute(email, message, threadId);
+        
         replyText = chatResponse.getReplyText();
-        Card card = CardResponse.createCardResponse(replyText);
-        reply.setCards(Collections.singletonList(card));
+        ActionType actionType = chatResponse.getActionType();
+        String documentUrl = chatResponse.getDocumentUrl();
+
+        // Create card if appropriate
+        if (actionType != null && documentUrl != null) {
```

Comment:
I like having the logic to only return a card when we need one! I had cards that just returned text for the Resource Helper bot but it looked kind of strange.

Reply:
+1 maybe we can have a constant list of action types for which we display cards? Anything that generates some form of structured information, like a list of values from a spreadsheet.

The link to the spreadsheet is a great feature, but I think it should be controlled by the macro creator, sometimes you may not want to disclose to the users that the data is stored in a specific file.